src/jsp/com/cubrid/plcsql/compiler/ParseTreeConverter.java

```diff
@@ -0,0 +1,1912 @@
+    public Expr visitPower_exp(Power_expContext ctx) {
+        Expr l = visitExpression(ctx.unary_expression(0), "Integer");
+        Expr r = visitExpression(ctx.unary_expression(1), "Integer");
+        return new ExprBinaryOp("Power", l, r);
+    }
+
+    @Override
+    public Expr visitNeg_exp(Neg_expContext ctx) {
+        Expr o = visitExpression(ctx.unary_expression(), "Integer");
+
+        Expr ret =
+                ctx.PLUS_SIGN() != null
+                        ? o
+                        : ctx.MINUS_SIGN() != null ? new ExprUnaryOp("Neg", o) : null;
+        assert ret != null;
+
+        return ret;
+    }
+
+    @Override
+    public Expr visitDate_exp(Date_expContext ctx) {
+        String s = ctx.quoted_string().getText();
+        s = quotedStrToJavaStr(s);
+        LocalDate date = DateTimeParser.DateLiteral.parse(s);
+        assert date != null : "invalid DATE string: " + s;
+        // System.out.println("[temp] date=" + date);
+        return new ExprDate(date);
+    }
+
+    @Override
+    public Expr visitTime_exp(Time_expContext ctx) {
+        String s = ctx.quoted_string().getText();
+        s = quotedStrToJavaStr(s);
+        LocalTime time = DateTimeParser.TimeLiteral.parse(s);
+        assert time != null : "invalid TIME string: " + s;
+        // System.out.println("[temp] time=" + time);
+        return new ExprTime(time);
+    }
+
+    @Override
+    public Expr visitTimestamp_exp(Timestamp_expContext ctx) {
+        String s = ctx.quoted_string().getText();
+        return parseZonedDateTime(s, false, "TIMESTAMP");
+    }
+
+    @Override
+    public Expr visitDatetime_exp(Datetime_expContext ctx) {
+        String s = ctx.quoted_string().getText();
+        s = quotedStrToJavaStr(s);
+        LocalDateTime datetime = DateTimeParser.DatetimeLiteral.parse(s);
+        assert datetime != null : "invalid DATETIME string: " + s;
+        // System.out.println("[temp] datetime=" + datetime);
+        return new ExprDatetime(datetime);
+    }
+
+    @Override
+    public Expr visitTimestamptz_exp(Timestamptz_expContext ctx) {
+        String s = ctx.quoted_string().getText();
+        return parseZonedDateTime(s, false, "TIMESTAMPTZ");
+    }
+
+    @Override
+    public Expr visitTimestampltz_exp(Timestampltz_expContext ctx) {
+        String s = ctx.quoted_string().getText();
+        return parseZonedDateTime(s, false, "TIMESTAMPLTZ");
+    }
+
+    @Override
+    public Expr visitDatetimetz_exp(Datetimetz_expContext ctx) {
+        String s = ctx.quoted_string().getText();
+        return parseZonedDateTime(s, true, "DATETIMETZ");
+    }
+
+    @Override
+    public Expr visitDatetimeltz_exp(Datetimeltz_expContext ctx) {
+        String s = ctx.quoted_string().getText();
+        return parseZonedDateTime(s, true, "DATETIMELTZ");
+    }
+
+    @Override
+    public Expr visitUint_exp(Uint_expContext ctx) {
+        try {
+            BigInteger bi = new BigInteger(ctx.UNSIGNED_INTEGER().getText());
```

Comment:
Long (or long) type can cover unsigned integer type in Java. BigInteger is capable of holding far bigger numbers than Long. But Long type performs better than BigInteger.

Reply:
In this case, UNSIGNED_INTEGER token is not limited to 64-bit integers, but corresponds to arbitrarily big unsigned integers (please see its definition in PcsLexer.g4).
Anyway, this line now seems to redundant because it checks the format of unsigned integers which must be surely valid by UNSIGNED_INTEGER lexer rules.